### PR TITLE
fix: Autocomplete trigger changes

### LINKED ISSUE #15429: [Bug]-[350]:Random keystrokes trigger autocomplete to show up
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

1. When code on the editor has already been indented, using keyboard board shortcut causes the autocomplete pop-up to open up wherever the cursor is present
2. Pasting a code snippet triggers autocomplete to show up
3. Pressing CTRL + any character shows autocomplete 

### Steps To Reproduce

1. Use the following code in the editor and prettify the code from the settings
```
export default {
	transformAPI: () => {
		const users =  fetchUsers.data.forEach((user) => {
			user.dob = moment(user.dob).format('Do MMM YYYY');
			return user;
		})
		return users
	}
}
```
2. Prettify from the drop-down options present on the top right
3. Now use keyboard shortcut again and observe that the autocomplete pop-up opens up as seen in the video

https://www.loom.com/share/9c22f3e3dd944a8cad11748e2cd1fe31

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by arunvjn**
## Description
PR to fix autocomplete trigger
- Disables autocomplete trigger when ctrl/meta/alt is pressed except for (ctrl/meta)+space.
- Disable autocomplete inside strings and comments

#### PR fixes following issue(s)
Fixes #15429

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [ ] Jest
- [x] Cypress
>
#### Test Plan
- Keyboard shortcut check for autocomplete
- Copy paste action-based check
- autocomplete suggestions inside strings and comments

#### Issues raised during DP testing
https://github.com/appsmithorg/appsmith/pull/24025#issuecomment-1577062645
>
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [x] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [x] Cypress test cases have been added and approved by SDET/manual QA
- [x] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by ohansFavour — APPROVED** [review]
Looks good

**Comment by ramsaptami** [comment]
- [ ] Prettify using keyboard shortcut still brings up autocomplete at cursor position

**Comment by arunvjn** [comment]
> * [ ]  Prettify using keyboard shortcut still brings up autocomplete at cursor position

Windows use shift+alt+f to prettify and wasn't handled earlier. You can now verify the fix at https://appsmith-67vbxwdxv-get-appsmith.vercel.app/

**Comment by ramsaptami** [comment]
LGTM @arunvjn

**Review by ohansFavour — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Autocomplete/Autocomplete_JS_spec.ts [hunk 1]
```diff
@@ -421,4 +421,81 @@ describe("Autocomplete tests", () => {
       );
     });
   });
+
+  it.only("10. Bug #10115 Autocomplete needs to show async await keywords instead of showing 'no suggestions'", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Autocomplete/Autocomplete_JS_spec.ts:425]: Remove `.only`. Move case to different spec.

**Review by Aishwarya-U-R — COMMENTED** [review]
Otherwise cypress looks good.

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**